host/libhackrf/src/hackrf.c

```diff
@@ -754,6 +759,36 @@ int ADDCALL hackrf_open_by_serial(
 	return hackrf_open_setup(usb_device, device);
 }
 
+int ADDCALL hackrf_open_by_fd(
+	int fd,
+	hackrf_device** device)
+{
+#ifndef _WIN32
+	libusb_device_handle* usb_device;
+
+	if (fd < 0) {
+		return HACKRF_ERROR_INVALID_PARAM;
+	}
+
+	if (device == NULL) {
+		return HACKRF_ERROR_INVALID_PARAM;
+	}
+
+	int err = libusb_wrap_sys_device(g_libusb_context, (intptr_t)fd, &usb_device);
```

Comment:
This function was added fairly recently (in libusb 1.0.23) so I think we should add an ifdef on `LIBUSB_API_VERSION` and return unsupported otherwise, to allow for building with older versions.